src/react/elements.js

```diff
@@ -92,6 +92,9 @@ function createPropsObject(
       }
       if (children !== realm.intrinsics.undefined) {
         setProp("children", children);
+        if (children instanceof ObjectValue || children instanceof AbstractObjectValue) {
+          children.makeFinal();
```

Comment:
This looks wrong to me. How do we know these children are created by React, and not passed in?

I would expect that we'd mark them as final early—when we create an array ([here](https://github.com/facebook/prepack/blob/fba61624ea064e92793213b80c6fe3364de5efec/src/evaluators/JSXElement.js#L167) and [here](https://github.com/facebook/prepack/blob/fba61624ea064e92793213b80c6fe3364de5efec/src/intrinsics/fb-www/react-mocks.js#L518)). If we don't create children ourselves then we shouldn't be marking them.